ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -1,66 +1,42 @@
+import { PrimaryType } from '../../../../../../../../shared/constants/signatures';
 import { parseTypedDataMessage } from '../../../../../../../../shared/modules/transaction.utils';
-import useTokenExchangeRate from '../../../../../../../components/app/currency-input/hooks/useTokenExchangeRate';
-import Name from '../../../../../../../components/app/name/name';
-import { Box, Text } from '../../../../../../../components/component-library';
-import Tooltip from '../../../../../../../components/ui/tooltip';
 import {
-  BackgroundColor,
-  BlockSize,
-  BorderRadius,
   Display,
-  TextAlign,
+  FlexDirection,
 } from '../../../../../../../helpers/constants/design-system';
-import { shortenString } from '../../../../../../../helpers/utils/util';
 import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
 import { currentConfirmationSelector } from '../../../../../../../selectors';
 import { SignatureRequestType } from '../../../../../types/confirm';
-import { IndividualFiatDisplay } from '../../../../simulation-details/fiat-display';
-import {
-  formatAmount,
-  formatAmountMaxPrecision,
-} from '../../../../simulation-details/formatAmount';
 import StaticSimulation from '../../shared/static-simulation/static-simulation';
+import PermitSimulationValueDisplay from './value-display/value-display';
 
-const PermitSimulation: React.FC<{
-  tokenDecimals: number;
-}> = ({ tokenDecimals }) => {
+const PermitSimulation: React.FC<object> = () => {
   const t = useI18nContext();
   const currentConfirmation = useSelector(
     currentConfirmationSelector,
   ) as SignatureRequestType;
 
+  const msgData = currentConfirmation.msgParams?.data;
   const {
     domain: { verifyingContract },
-    message: { value },
-  } = parseTypedDataMessage(currentConfirmation.msgParams?.data as string);
-
-  const exchangeRate = useTokenExchangeRate(verifyingContract);
+    message: { details, value },
+    primaryType,
+  } = parseTypedDataMessage(msgData as string);
 
-  const fiatValue = useMemo(() => {
-    if (exchangeRate && value) {
-      const tokenAmount = calcTokenAmount(value, tokenDecimals);
-      return exchangeRate.times(tokenAmount).toNumber();
-    }
-    return undefined;
-  }, [exchangeRate, value]);
+  const isPermitSingle = primaryType === PrimaryType.PermitSingle;
+  const isPermitBatch = primaryType === PrimaryType.PermitBatch;
 
-  const { tokenValue, tokenValueMaxPrecision } = useMemo(() => {
-    if (!value) {
-      return { tokenValue: null, tokenValueMaxPrecision: null };
-    }
+  let tokenDetails;
 
-    const tokenAmount = calcTokenAmount(value, tokenDecimals);
-
-    return {
-      tokenValue: formatAmount('en-US', tokenAmount),
-      tokenValueMaxPrecision: formatAmountMaxPrecision('en-US', tokenAmount),
-    };
-  }, [tokenDecimals, value]);
+  if (isPermitSingle) {
+    tokenDetails = [details];
+  } else if (isPermitBatch) {
+    tokenDetails = details;
+  }
```

Comment:
If we can create array in both cases, that may simplify code. For `isPermitSingle` it will be array with only 1 element.